tempodb/encoding/vparquet4/schema.go

```diff
@@ -0,0 +1,963 @@
+	}
+	// the two below are used in tag name search. they only include
+	//  custom attributes that are mapped to parquet "special" columns
+	traceqlResourceLabelMappings = map[string]string{
+		LabelServiceName:      "rs.list.element.Resource.ServiceName",
+		LabelCluster:          "rs.list.element.Resource.Cluster",
+		LabelNamespace:        "rs.list.element.Resource.Namespace",
+		LabelPod:              "rs.list.element.Resource.Pod",
+		LabelContainer:        "rs.list.element.Resource.Container",
+		LabelK8sClusterName:   "rs.list.element.Resource.K8sClusterName",
+		LabelK8sNamespaceName: "rs.list.element.Resource.K8sNamespaceName",
+		LabelK8sPodName:       "rs.list.element.Resource.K8sPodName",
+		LabelK8sContainerName: "rs.list.element.Resource.K8sContainerName",
+	}
+	traceqlSpanLabelMappings = map[string]string{
+		LabelHTTPMethod:     "rs.list.element.ss.list.element.Spans.list.element.HttpMethod",
+		LabelHTTPUrl:        "rs.list.element.ss.list.element.Spans.list.element.HttpUrl",
+		LabelHTTPStatusCode: "rs.list.element.ss.list.element.Spans.list.element.HttpStatusCode",
+	}
+
+	parquetSchema = parquet.SchemaOf(&Trace{})
+)
+
+// droppedAttrCounter represents an entity that can count dropped attributes
+type droppedAttrCounter interface {
+	addDroppedAttr(n int32)
+}
+
+type Attribute struct {
+	Key string `parquet:",snappy,dict"`
+
+	ValueType    AttrType  `parquet:",snappy,delta"`
+	Value        []string  `parquet:",snappy,dict,list"`
+	ValueInt     []int64   `parquet:",snappy,list"`
+	ValueDouble  []float64 `parquet:",snappy,list"`
+	ValueBool    []bool    `parquet:",snappy,list"`
+	ValueDropped string    `parquet:",snappy,optional"`
+}
+
+// DedicatedAttributes add spare columns to the schema that can be assigned to attributes at runtime.
+type DedicatedAttributes struct {
+	String01 *string `parquet:",snappy,optional,dict"`
+	String02 *string `parquet:",snappy,optional,dict"`
+	String03 *string `parquet:",snappy,optional,dict"`
+	String04 *string `parquet:",snappy,optional,dict"`
+	String05 *string `parquet:",snappy,optional,dict"`
+	String06 *string `parquet:",snappy,optional,dict"`
+	String07 *string `parquet:",snappy,optional,dict"`
+	String08 *string `parquet:",snappy,optional,dict"`
+	String09 *string `parquet:",snappy,optional,dict"`
+	String10 *string `parquet:",snappy,optional,dict"`
+}
+
+type Event struct {
+	TimeSinceStartNano     uint64      `parquet:",delta"`
+	Name                   string      `parquet:",snappy,dict"`
+	Attrs                  []Attribute `parquet:",list"`
+	DroppedAttributesCount int32       `parquet:",snappy,delta"`
+}
+
+func (e *Event) addDroppedAttr(n int32) {
+	e.DroppedAttributesCount += n
+}
+
+type Link struct {
+	TraceID                []byte      `parquet:","`
+	SpanID                 []byte      `parquet:","`
+	TraceState             string      `parquet:",snappy"`
+	Attrs                  []Attribute `parquet:",list"`
+	DroppedAttributesCount int32       `parquet:",snappy,delta"`
```

Comment:
i'm confused by this dropped attributes tracking. if we store it and return it faithfully on request by trace id then i don't think we need to track dropped anything.

these attrs would be invisible to traceql but we can just clearly document what we support

Reply:
* Unsupported attributes values that are stored in the `ValueDropped` columns are not returned by the API and appear as dropped. Therefore the incrementing the dropped attribute count for those values is appropriate
* `DroppedAttributesCount` may also be > 0 when attributes are removed by other parts of the tracing pipeline

Reply:
going to mark this resolve and consolidate dropped attributes discussion above.